storage/postgres/postgres_test.go

```diff
@@ -23,20 +23,36 @@ var store *Storage
 var userDataMock = "bar"
 
 func TestMain(m *testing.M) {
-	var err error
-	var c dockertest.ContainerID
-	c, db, err = dockertest.OpenPostgreSQLContainerConnection(15, time.Millisecond * 500)
+	c, err := dockertest.ConnectToPostgreSQL(15, time.Second, func(url string) bool {
+		var err error
+		dbTemp, err := sql.Open("postgres", url)
+		if err != nil {
+			return false
+		}
+		store = New(dbTemp)
+		if err = store.CreateSchemas(); err != nil {
+			log.Fatalf("Could not ping database: %v", err)
+		}
+
+		db = dbTemp
+
+		return db.Ping() == nil
+	})
+
 	if err != nil {
-		log.Fatalf("Could not set up PostgreSQL container: %v", err)
+		log.Fatalf("Could not connect to database: %s", err)
 	}
-	defer c.KillRemove()
 
-	store = New(db)
-	if err = store.CreateSchemas(); err != nil {
-		log.Fatalf("Could not set up schemas: %v", err)
-	}
+	retCode := m.Run()
+
+	// force teardown
+	tearDown(c)
+
+	os.Exit(retCode)
+}
 
-	os.Exit(m.Run())
```

Comment:
Oh ok the problem is here:

```
	c, err := dockertest.ConnectToPostgreSQL(15, time.Second, func(url string) bool {
		var err error
		dbTemp, err := sql.Open("postgres", url)
		if err != nil {
			return false
		}
		store = New(dbTemp)
		if err = store.CreateSchemas(); err != nil {
			log.Fatalf("Could not ping database: %v", err)
		}

		db = dbTemp

		return db.Ping() == nil
	})
```

The callback method is just for checking if the database is already responsive. Depending on the environment (e.g. travis VM), docker images need a little time before they go responsive - although they are booted already! An unresponsive docker image (e.g. postgres cleaning up -> then listening) is solved by checking if the database is pingable. If it's not, dockertest retries connecting.

Therefore, CreateSchemas must not be inside of that callback function but instead below it. I will fix this and ping you in the PR.

My mistake for not catching that during review.